build.yaml

```diff
@@ -168,14 +196,28 @@ common_cli:
     # https://gitlab.com/gitlab-org/gitlab-ci-multi-runner/issues/20
     # So we will place the virtualenv in /tmp/v instead
     #
-    sudo rm -rf \$venv
-    sudo rm -f results.xml
-    sudo rm -f trial.log
-    sudo rm -rf _trial_temp/
+    if [ "\$(is_osx)" == "true" ]; then
+      # OSX doesn't allow us to sudo (no tty)
+      rm -f results.xml
+      rm -f trial.log
+      rm -rf _trial_temp/
+      rm -rf \${venv}
```

Comment:
Can you expand on that somewhat (and add a comment explaining it).

It doesn't explain why there is also the cleanup on centos/ubuntu, which presumably does have the pre-scm plugin run.

In fact, is this even needed on OSX? My understanding is that this was necessary to cleanup root-owned files, and that there is already a step that does `git clean` making this unnecessary on OSX.